WordPress/src/main/java/org/wordpress/android/ui/posts/services/PostUploadService.java

```diff
@@ -595,13 +563,30 @@ public void onPostUploaded(OnPostUploaded event) {
                     false);
             mFirstPublishPosts.remove(event.post.getId());
         } else {
-            // TODO: MediaStore?
-            // WordPress.wpDB.deleteMediaFilesForPost(mPost);
             mPostUploadNotifier.cancelNotification(event.post);
             boolean isFirstTimePublish = mFirstPublishPosts.remove(event.post.getId());
             mPostUploadNotifier.updateNotificationSuccess(event.post, site, isFirstTimePublish);
         }
 
         finishUpload();
     }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onMediaUploaded(OnMediaUploaded event) {
+        // Event for unknown media, ignoring
+        if (event.media == null || mMediaLatchMap.get(event.media.getId()) == null) {
+            AppLog.w(T.POSTS, "Media event not recognized: " + event.media);
+            return;
+        }
+
+        if (event.completed) {
+            AppLog.i(T.POSTS, "Media upload completed for post. Media id: " + event.media.getId()
+                    + ", post id: " + mCurrentUploadingPost.getId());
+            mMediaLatchMap.get(event.media.getId()).countDown();
+            mMediaLatchMap.remove(event.media.getId());
+        } else if (!event.canceled && !event.isError()) {
+            mPostUploadNotifier.updateNotificationProgress(mCurrentUploadingPost, event.progress);
+        }
```

Comment:
We should cancel the post upload here in case of a media upload error (I'm pretty sure this was the previous behavior).